src/glue/jobs/compare_parquet_datasets.py

```diff
@@ -0,0 +1,391 @@
+import os
+import json
+import logging
+import argparse
+
+import boto3
+import datacompy
+import pandas as pd
+import synapseclient
+from pyarrow import fs
+import pyarrow.parquet as pq
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+INDEX_FIELD_MAP = {
+    "dataset_enrolledparticipants": ["ParticipantIdentifier"],
+    "dataset_fitbitprofiles": ["ParticipantIdentifier", "ModifiedDate"],
+    "dataset_fitbitdevices": ["ParticipantIdentifier", "Date"],
+    "dataset_fitbitactivitylogs": ["LogId"],
+    "dataset_fitbitdailydata": ["ParticipantIdentifier", "Date"],
+    "dataset_fitbitintradaycombined": ["ParticipantIdentifier", "Type", "DateTime"],
+    "dataset_fitbitrestingheartrates": ["ParticipantIdentifier", "Date"],
+    "dataset_fitbitsleeplogs": ["LogId"],
+    "dataset_healthkitv2characteristics": ["HealthKitCharacteristicKey"],
+    "dataset_healthkitv2samples": ["HealthKitSampleKey"],
+    "dataset_healthkitv2samples_deleted": ["HealthKitSampleKey"],
+    "dataset_healthkitv2heartbeat": ["HealthKitHeartbeatSampleKey"],
+    "dataset_healthkitv2statistics": ["HealthKitStatisticKey"],
+    "dataset_healthkitv2clinicalrecords": ["HealthKitClinicalRecordKey"],
+    "dataset_healthkitv2electrocardiogram": ["HealthKitECGSampleKey"],
+    "dataset_healthkitv2workouts": ["HealthKitWorkoutKey"],
+    "dataset_healthkitv2activitysummaries": ["HealthKitActivitySummaryKey"],
+    "dataset_googlefitsamples": ["GoogleFitSampleKey"],
+    "dataset_symptomlog": ["DataPointKey"],
+}
+
+
+def read_args():
+    parser = argparse.ArgumentParser(
+        description=(
+            "Compare parquet datasets between two namespaced S3 bucket locations"
+        )
+    )
+    parser.add_argument(
+        "--staging-namespace",
+        required=True,
+        help="The name of the staging namespace to use",
+    )
+    parser.add_argument(
+        "--main-namespace",
+        required=True,
+        help=("The name of the main namespace to use"),
+    )
+    parser.add_argument(
+        "--parquet-bucket",
+        required=True,
+        help=("The name of the S3 bucket containing the S3 files to compare"),
+    )
+    args = parser.parse_args()
+    return args
+
+
+def get_duplicated_columns(dataset: pd.DataFrame) -> list:
+    """Gets a list of duplicated columns in a dataframe"""
+    return dataset.columns[dataset.columns.duplicated()].tolist()
+
+
+def has_common_cols(staging_dataset: pd.DataFrame, main_dataset: pd.DataFrame) -> list:
+    """Gets the list of common columns between two dataframes"""
+    common_cols = staging_dataset.columns.intersection(main_dataset.columns).tolist()
+    return common_cols != []
+
+
+def get_missing_cols(staging_dataset: pd.DataFrame, main_dataset: pd.DataFrame) -> list:
+    """Gets the list of missing columns present in main but not in staging"""
+    missing_cols = main_dataset.columns.difference(staging_dataset.columns).tolist()
+    return missing_cols
```

Comment:
Just yesterday, I found that datacompy.Compare has functions that handles some of the above (e.g: gets unique columns in one dataset but not the other here: https://github.com/capitalone/datacompy/blob/develop/datacompy/core.py#L220-L230. It's marked as a TODO because I feel like this PR has blown up already so much and having those changes is a bit more involved.